uniffi_bindgen/src/bindings/python/templates/ErrorTemplate.py

```diff
@@ -1,49 +1,91 @@
-class RustError(ctypes.Structure):
+class InternalError(Exception):
+    pass
+
+class RustCallStatus(ctypes.Structure):
     _fields_ = [
-        ("code", ctypes.c_int32),
-        ("message", ctypes.c_void_p),
+        ("code", ctypes.c_int8),
+        ("error_buf", RustBuffer),
     ]
 
-    def free(self):
-        rust_call_with_error(InternalError, _UniFFILib.{{ ci.ffi_string_free().name() }}, self.message)
+    # These match the values from the uniffi::rustcalls module
+    CALL_SUCCESS = 0
+    CALL_ERROR = 1
+    CALL_PANIC = 2
 
     def __str__(self):
-        return "RustError(code={}, message={})".format(
-            self.code,
-            str(ctypes.cast(self.message, ctypes.c_char_p).value, "utf-8"),
-        )
+        if self.code == RustCallStatus.CALL_SUCCESS:
+            return "RustCallStatus(CALL_SUCCESS)"
+        elif self.code == RustCallStatus.CALL_ERROR:
+            return "RustCallStatus(CALL_ERROR)"
+        elif self.code == RustCallStatus.CALL_PANIC:
+            return "RustCallStatus(CALL_SUCCESS)"
+        else:
+            return "RustCallStatus(<invalid code>)"
+{%- for e in ci.iter_error_definitions() %}
 
-class InternalError(Exception):
-    @staticmethod
-    def raise_err(code, message):
-        raise InternalError(message)
-
-{% for e in ci.iter_error_definitions() %}
 class {{ e.name()|class_name_py }}:
-    {%- for value in e.values() %}
-    class {{ value|class_name_py }}(Exception):
-        pass
+    # Each variant is a nested class of the error itself.
+    {%- for variant in e.variants() %}
+
+    class {{ variant.name()|class_name_py }}(Exception):
+        def __init__(self{% for field in variant.fields() %}, {{ field.name()|var_name_py }}{% endfor %}):
+            {%- if variant.has_fields() %}
+            {%- for field in variant.fields() %}
+            self.{{ field.name()|var_name_py }} = {{ field.name()|var_name_py }}
+            {%- endfor %}
+            {% else %}
+            pass
+            {%- endif %}
+
+        def __str__(self):
+            return "{{ e.name()|class_name_py }}.{{ variant.name()|class_name_py }}({% for field in variant.fields() %}{{ field.name() }}={}{% if !loop.last %},{% endif %}{% endfor %})".format({% for field in variant.fields() %}self.{{ field.name() }}{% if !loop.last %},{% endif %}{% endfor %})
```

Comment:
yikes, but also, probably not that much value in trying to split this over multiple lines..,